app/decorators/fields_definition_decorator.rb

```diff
@@ -6,9 +6,10 @@ class FieldsDefinitionDecorator < ApplicationDecorator
   def new_application_data(provider)
     type = field_type(provider)
     input_name = "#{target.downcase}#{type == 'extra' ? '[extra_fields]' : ''}[#{name}]"
+    required_value = name != 'description' # Set required to false for the description field
```

Comment:
The issue is actually deeper than this...
This fix just masks the input label on the UI (which is not right).

Actually, the `description` field is a built-in field marked as required by default, and there is no way to change it:

![image](https://github.com/3scale/porta/assets/1270328/5d042a2a-eae7-4db2-b7b6-3d36ffe30d0f)


https://github.com/3scale/porta/blob/2f20448547aea8139ff2dd6a966329255d7b8de6/app/models/cinstance.rb#L48

This seems to have been like that forever ([since 12 years ago](https://github.com/3scale/system/blame/3213fb8ef08e86df86ccf837517d30b3f446fd4c/app/models/cinstance.rb#L44)).

There is also a validation here: https://github.com/3scale/porta/blob/2f20448547aea8139ff2dd6a966329255d7b8de6/app/models/cinstance.rb#L83
But it needs to be investigated whether this still makes sense (i.e. figure out why the description was made a required field in the first place).

Finally, if we make the "description" optional, apart from all of the code above, we need to update the API docs [here](https://github.com/3scale/porta/blob/master/doc/active_docs/account_management_api.json#L9801).

Reply:
Okay  let me check this.

Reply:
I think with these we also need to change the object in DB to required false to show it as not required on filed definition page the screenshot which you shared, which doesn't sounds like a solution,

Although I am able to create application without description, but this will confuse if it will show required in the the account filed definition.

the required changes are done and pushed from my end. I don't the reason why the made this required but these are the by default field definitions so that's why may be they made it required. But I think now it won't be for new users may be.


![Screenshot from 2023-09-25 15-39-46](https://github.com/3scale/porta/assets/84672731/2a225e19-a7c7-4024-8856-9df46fc97b21)


`FieldsDefinition id: 13, account_id: 2, created_at: "2023-09-11 10:44:25", updated_at: "2023-09-11 10:44:25", target: "Cinstance", hidden: false, required: true, label: "Description", name: "description", choices: [], hint: nil, read_only: false, pos: 2, tenant_id: 2>`

Reply:
Well, I think we need to try to figure out in what cases the description would actually be required. Because if we just remove the validation with no understanding, we may break something.

Also, with the other changes, the change in this line (in the decorator) might not be needed. Please check.

Reply:
@mayorova After checking in code and old commit I haven't find any solid reason, but looking at the failing test which are also failing because from API docs [here](https://github.com/3scale/porta/blob/master/doc/active_docs/account_management_api.json#L9801). description is removed.

All I found is we are using it on very few places on code and one with no check is on stats view page
https://github.com/3scale/porta/blob/2f20448547aea8139ff2dd6a966329255d7b8de6/app/lib/stats/views/usage.rb#L46 here.

which also not looks breakable . If sounds good to you I can fix test and then cpmplete this PR .

Reply:
I looked into this issue and it has some history... See [this issue](https://github.com/3scale/system/issues/1889) and [this fix](https://github.com/3scale/system/commit/a88f574a0a801ddc80a030f26201058dca574414)

So, historically, the "description" field for Cinstance (application) was only required for some conditions:
- multiple applications per service are allowed, by making the corresponding feature "Visible":
![image](https://github.com/3scale/porta/assets/1270328/d74f109a-c094-4a61-b3b9-762e819fc192)
OR
- `intentions_required` is set to `true` on the service. This is configured by this checkbox in the Service Usage Rules:
![image](https://github.com/3scale/porta/assets/1270328/7861e2dc-2fe4-4da6-a455-2e27561ac9d6)
The two are kind of mutually exclusive (i.e. you don't see the `intentions_required` checkbox if multiple applications are visible).

So, in some conditions users could created applications with no descriptions. BUT, later, when the configuration changed, such applications could become invalid (because under new conditions the description was required), so actions such as plan change couldn't be performed.

So the solution that was applied: only validate the "description" field when an action is triggered by the user (i.e. creating an app from the UI or via API).

IMO, this is very strange, but I guess this was done so to avoid breaking changes.

The similar thing was done to the "name" field, see [issue](https://github.com/3scale/system/issues/6351) and [PR](https://github.com/3scale/system/pull/6444)

By the way, there is [a comment](https://github.com/3scale/system/pull/6444#issuecomment-191188006) on the last PR:

> The description field is currently required for multi app accounts and optional for single app accounts. It's also listed as a required attribute for the API. **I suggest we make it an optional attribute for all.** But that would be outside of the scope of this issue

I fully agree with the suggestion and I think this is what we need to do.

Also, I'd say we need to review and ideally get rid of this `@validate_human_edition` thing. I think, the validation rules should be the same for all kinds of interactions, and I believe that the applications that currently do not have any name are very old legacy apps that probably nobody uses anymore. But that's for another PR.